schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroSchemaProvider.java

```diff
@@ -106,14 +108,24 @@ public String normalize(Schema schema) throws IOException {
     // Added default and aliases handling here.
     private static Appendable build(Map<String, String> env,
                                     Schema schema,
-                                    Appendable appendable) throws IOException {
+                                    Appendable appendable) throws IOException, InvalidSchemaException {
         boolean firstTime = true;
         Schema.Type schemaType = schema.getType();
         String fullName = schema.getFullName();
         switch (schemaType) {
-            default: // boolean, bytes, double, float, int, long, null, string
+            default: // boolean, bytes, double, float, int, long, null
                 return appendable.append('"').append(schemaType.getName()).append('"');
 
+            case STRING:
+                String stringType = schema.getProp(GenericData.STRING_PROP);
+                if ( stringType == null )
+                    stringType = schemaType.getName();
```

Comment:
When `GenericData.STRING_PROP` does not exist, the value is considered as `Utf8`. 
string type
- without `GenericData.STRING_PROP`
- with `GenericData.STRING_PROP` as Utf8
should generate same finger print.

But the current logic generates different strings in the above scenarios. So, it is better to handle default value for `GenericData.STRING_PROP` and generate equivalent strings. One way to fix this is always have `schemaType.getName()` and append effective value for `GenericData.STRING_PROP` property.
Generated string can be of the form like `{"string", "$javaType"}`, javaType can be CharSequence, String and Utf8.

Good to add tests to check whether it generates equivalent fingerprints for equivalent effective types respectively.